catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Clients.java

```diff
@@ -19,6 +19,8 @@
 import org.projectnessie.catalog.secrets.BasicCredentials;
 import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
 import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
```

Comment:
The reason why we do not have the `DefaultCredentialsProvider` here is that it can run for a really long time. This code here would run for every access to the credentials.

If we want to infer data from EC2 metadata, that should IMHO only happen for the default settings and only be done once.

Reply:
Or maybe we could memoize the resolved credentials per bucket? That would also allow e.g. to print a warning message without spamming the logs.

Reply:
No memoizing here, please.
We also need to consider how this change could conflict in case a secrets manager/vault is being use - e.g. when the "right" secret is added later.
Let's not rush this.

Reply:
Fair enough. That said I don't think each call to this method will pay the penalty of traversing the whole supplier chain because `reuseLastProviderEnabled` is true by default.